docs/design/api/storage.rst

```diff
@@ -0,0 +1,445 @@
+
+        {
+          "data": {
+            "valid_until": 1437208555580
+          }
+        }
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 Ok
+
+        {
+          "data": {
+            "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+            "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+            "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+            "added_on": 1432024555580,
+            "valid_until": 1437208555580
+          },
+          "permissions": {
+            "read": ["fxa:5331be33303ccff19d3c49b6da276913",
+                     "hawk:144c22a77d75937740ec3c957fbdb1d1"]
+          }
+        }
+
+    :statuscode 200: Ok, no error
+
+
+As the payments application I want to be able to alter all purchases
+====================================================================
+
+You'll need to do a BATCH operation with all the sub-operations in there.
+
+- First get the list of records you want to modify.
+
+.. http:get:: /buckets/(bucket_id)/collections/(collection_id)/records
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        GET /buckets/mozilla/collections/payments/records?seller=hawk:144c22a77d75937740ec3c957fbdb1d1 HTTP/1.1
+        Authorization: Basic <PaymentApp Basic Auth credentials>
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+
+        {
+          "items": [
+            {
+                "id": "dc86afa9-a839-4ce1-ae02-3d538b75496f",
+                "last_modified": 1432024555580,
+                "buyer": "fxa:5331be33303ccff19d3c49b6da276913",
+                "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+                "product": "bb5bf35f-cb2b-40e7-b1ef-8b097bd550f2",
+                "added_on": 1432024555580,
+                "valid_until": 1463560555580
+            },
+            {
+                "id": "db4d95e1-c076-4848-950c-cf462b2631f0",
+                "last_modified": 1430140411480,
+                "title": "MoFo",
+                "buyer": "fxa:465afecea6b565c85fd980a603747fec",
+                "seller": "hawk:144c22a77d75937740ec3c957fbdb1d1",
+                "product": "ecd68f3c-984b-471c-a670-8411e5247358",
+                "added_on": 1430140411480,
+                "valid_until": 1432732411480
+            }
+          ]
+        }
+
+    :query seller: Filter on the seller app identifier
+    :statuscode 200: Ok, no error
+
+
+- Then run a BATCH request:
+
+If you want to add the ``read`` permission for the seller app to all
+records of the app, you could use:
+
+.. http:post:: /batch
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        POST /batch HTTP/1.1
+        Authorization: Basic <PaymentApp Basic Auth credentials>
+
+        {
+          "defaults": {
+            "data": {
+              "permissions": {
+                "read": [
+                  "+hawk:144c22a77d75937740ec3c957fbdb1d1"
```

Comment:
What I meant earlier was that the plus sign here is cryptic. What does it do? Instead of prefixing the ID with a plus sign could we represent the request in a more specific way with words? I don't know what it does so I can't suggest something ;) Roughly: ``{"read": [{"some_modifier": "hawk:144c22a77d75937740ec3c957fbdb1d1"}]}`` (as you see this does not use a plus sign)

However, if the plus sign is buried deep inside some external library which is hard to change then I understand.